tensornetwork/tensornetwork.py

```diff
@@ -317,8 +318,26 @@ def update_axis(self, old_axis: int, old_node: Node, new_axis: int,
                        "node1: '{}', axis1: {}, node2: '{}', axis2: {}".format(
                            self, old_node, old_axis, self.node1, self.axis1,
                            self.node2, self.axis2))
-
-  def is_dangling(self):
+  @property
+  def node1(self) -> Node:
+    val = self._node1()
+    if val is None:
+      raise ValueError("node1 for edge '{}' no longer exists.".format(self))
+    return val
+
+  @property
+  def node2(self) -> Optional[Node]:
+    return self._node2() if self._node2 else None  
```

Comment:
It looks like this makes it impossible to distinguish between a dangling edge and an edge with a garbage-collected neighbor. WDYT about introducing a boolean flag to indicate whether the edge is dangling so that we don't have to rely on special value `None` for this purpose?

Reply:
That seems fine. I'll add it.